---
title: "Metabolome paper - Figure 5D-F and Supplementary Figure 5"
author: "Viviana Freire Zapata"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

# Loading libraries

```{r message=FALSE, warning=FALSE}
library(tidyverse)
library(readxl)
library(patchwork)
library(picante)
library(ggpubr)
library(treedataverse)
library(ggalluvial)
library(ggthemes)
```

# Load data

## Taxonomy data

```{r}
tax <- read_csv('../input/network_nodes_taxonomy.csv') %>% 
  mutate(Phylum = str_remove(Phylum, 'p__'))

phylum_list <- sort(unique(tax$Phylum))
```

## Data for Sankey diagrams

```{r}
sankey_files <- list.files(path = '../input', pattern = 'Sankey', 
                           full.names = TRUE)
names(sankey_files) <- c('Bog', 'Fen', 'Palsa')

sankey_df <- imap(sankey_files, function(file, y){
  df <- read_tsv(file, col_names = FALSE) %>% 
    rename(Taxa = X1,
           Reaction = X2,
           Freq = X3) %>% 
    mutate(Category = case_when(str_detect(Reaction, 'C-S') ~ 'Carbon',
                                str_detect(Reaction, 'N-S') ~ 'Nitrogen',
                                str_detect(Reaction, 'S-S') ~ 'Sulfur',
                                str_detect(Reaction, 'O-S') ~ 'Others'),
           Category = factor(Category, levels = c('Carbon', 'Nitrogen', 'Sulfur', 'Others')),
           Taxa = ifelse(str_detect(Taxa, 'proteobacteria'), 'Proteobacteria', Taxa),
           Taxa = factor(Taxa, levels = phylum_list),
           Habitat = y,
           Habitat = factor(Habitat, levels = c('Palsa', 'Bog', 'Fen'))
    )
})
```

## Bacterial phylogenetic tree

```{r}
tree <- read.tree('../Assembly_feature_level_microbiome/input/bac_arc_tree_comb.tree')
```

## Bacteria in networks

```{r}
network_otu <- read_csv('tables/networked_bacteria.csv') %>% 
  rename(old_ID = OTU) 

ids <-  network_otu %>% 
  select(Name, old_ID) %>% 
  distinct()
```

## CAZymes

```{r}
from_dram <- read_tsv('../input/1864_97percentmags_annotations_combined.tsv') %>% 
  filter(!is.na(cazy_id))
```

## metaT

```{r}
metaT_all <- read_csv('../input/Emerge_metaTs_processed_v7/metaT_pathway_filter_2012.csv') %>% 
  filter(pathway == 'CAZy',
         str_detect(SampleID__, 'MainAutochamber.2012'))
```

# Plot colors 

```{r}
phylum_colors <- setNames(ggpubr::get_palette('Paired', length(phylum_list)),
                          nm = phylum_list)

hab_colors <- c('Bog' = '#058000','Fen' = '#0001FF', 'Palsa' = '#703C1B')
```

# Figure 4

## Alluvial plots

```{r}
df_sankey_f <- do.call(rbind, sankey_df) %>% 
  mutate(Taxa = fct_lump_n(Taxa, 10, w = Freq))

legend_p <- data.frame(Taxa = c(names(phylum_colors), 'Other')) %>% 
  mutate(Taxa = fct_other(Taxa, keep = levels(df_sankey_f$Taxa))) %>% 
  ggplot() +
  geom_col(aes(x = Taxa,
               y = 1,
               fill = Taxa),
           alpha = 0.75) +
  scale_fill_manual(values = c(phylum_colors, "Other" = 'gray30')) +
  guides(fill = guide_legend(ncol = 6)) +
  theme_bw() +
  theme(legend.position = 'bottom',
        legend.text = element_text(size = 5),
        legend.title = element_text(size = 7),
        legend.key.size = unit(.75, 'line'))

legend <- get_legend(legend_p)

sankey_plots <- imap(sankey_df, function(df, y){
  plot <- df %>%
    mutate(Taxa = fct_other(Taxa, keep = levels(df_sankey_f$Taxa))) %>% 
    ggplot(aes(y = Freq,
               axis1 = Taxa,
               axis2 = Reaction,
               axis3 = Category)) +
    geom_alluvium(aes(fill = Taxa),
                  width = 1/12) +
    geom_stratum(width = 1/12, fill = "white", color = "black") +
    ggrepel::geom_label_repel(stat = "stratum", 
                              aes(label = after_stat(stratum)),
                              size = 1.35,
                              label.padding = 0.1,
                              show.legend = FALSE) +
    labs(title = y) +
    scale_x_discrete(limits = c("Phylum", "Reaction","Category"), expand = c(.08, .05)) +
    scale_fill_manual(values = c(phylum_colors, 'Other' = 'gray30'), 
                      na.value = 'white', drop = FALSE) +
    theme_bw() +
    theme(plot.title = element_text(face = 'bold', hjust = 0.5, size = 7),
          legend.position = "none",
          axis.text = element_text(size = 6),
          axis.title = element_text(size = 7))
  
  return(plot)
})
```

## Saving plot

```{r}
alluvial <- ((ggplot() + theme_void() | ggplot() + theme_void() | ggplot() + theme_void()) /
  (sankey_plots$Palsa | sankey_plots$Bog | sankey_plots$Fen) / legend) +
  plot_layout(heights = c(6, 5, 1)) +
  plot_annotation(tag_levels = list(c('A', 'B', 'C', 'D', 'E', 'F'))) &
  theme(plot.tag = element_text(size = 7))

ggsave('output/Figure_4D-F_sankey.png', alluvial, dpi = 300, width = 210, height = 180, units = 'mm')

```

# Supplementary Figure 5

## Filtering tree and adding phylogeny

```{r message=FALSE, warning=FALSE, error=FALSE}
# Filtering
bac_list_hab <- from_dram %>% 
  select(fasta) %>% 
  distinct() %>% 
  inner_join(network_otu, by = c('fasta' = 'old_ID'))

tips <- setNames(bac_list_hab$fasta,
                 nm = bac_list_hab$fasta)

matched_phylo <- match.phylo.data(tree, tips)

# Phylogeny

temp_annot <- as_tibble(matched_phylo$phy) %>% 
  left_join(ids, by = c('label' = 'old_ID')) %>% 
  left_join(tax , by = 'Name')
```

## Preparing CAZyme data

```{r}
cazymes_hits <- from_dram %>% 
  select(fasta, cazy_id) %>%
  filter(fasta %in% network_otu$old_ID,
         !is.na(cazy_id)) %>%
  group_by(fasta) %>% 
  summarise(num_cazy = n()) %>% 
  left_join(ids, by = c('fasta' = 'old_ID')) 

```

```{r}
expressed_cazy <- metaT_all %>% 
  select(genome, SampleID__, cazy_id) %>% 
  distinct() %>% 
  mutate(Habitat = case_when(str_detect(SampleID__, '_S_') ~ 'Bog',
                             str_detect(SampleID__, '_E_') ~ 'Fen',
                             TRUE ~ 'Palsa')) %>% 
  inner_join(network_otu, by = c('genome' = 'old_ID', 'Habitat')) %>% 
  group_by(genome, Habitat) %>% 
  count()
```


## Plotting tree

```{r}
annot_tree <- as.treedata(temp_annot)

plot_tree <- ggtree(annot_tree, layout = 'fan', open.angle = 25) +
  geom_tippoint(aes(color = Phylum)) +
  scale_color_manual(values = phylum_colors) +
  guides(color = guide_legend(ncol = 3)) +
  geom_fruit(data = bac_list_hab,
             geom = geom_tile,
             mapping = aes(y = fasta,
                           x = Habitat,
                           fill = Habitat),
             color = 'white',
             offset = .1,
             pwidth = 0.2) +
  scale_fill_manual(values = hab_colors) +
  ggnewscale::new_scale_fill() +
  geom_fruit(data = cazymes_hits,
             geom = geom_col,
             mapping = aes(y = fasta,
                           x = num_cazy,
                           fill = num_cazy),
             color = 'black',
             offset = 0.1,
             pwidth = 0.2,
             axis.params = list(axis = 'x',
                                text.size = 2,
                                text.angle = 270,
                                hjust = -1),
             grid.params = list()) +
  scale_fill_distiller('Number of CAZymes', 
                       palette = 'YlOrRd', direction = 1, na.value = 'white') +
  ggnewscale::new_scale_fill() +
  geom_fruit(data = expressed_cazy,
             geom = geom_col,
             mapping = aes(y = genome,
                           x = n,
                           fill = n),
             color = 'black',
             offset = 0.1,
             pwidth = 0.2,
             axis.params = list(axis = 'x',
                                text.size = 2,
                                text.angle = 270,
                                hjust = -1),
             grid.params = list()) +
  scale_fill_distiller('Number of expressed CAZymes', 
                       palette = 'Purples', direction = 1, na.value = 'white') +
  theme(legend.text = element_text(size = 10))

plot_tree

ggsave('output/Supplementary_fig_5.png', plot_tree, dpi = 300, height = 8, width = 12)
```



