---
title: "Metabolome paper - Supplementary Figure 2"
author: "Viviana Freire Zapata"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

# Load libraries

```{r message=FALSE, warning=FALSE}
library(readxl)
library(ggtext)
library(ggpubr)
library(tidyverse)

```

# Loading data

## MAG metadata

```{r}
metadata <- read_csv("../Assembly_feature_level_microbiome/input/metadata_matched_MAGs_data.csv")
```

## Matching genomes

```{r}
hab <- c('Palsa', 'Bog', 'Fen')
micro_data <- map(str_to_lower(hab), function(x){
  file <- paste0('../Metabolite_clusters/output/', x, '_sig_corr_taxa.csv')
  df <- read_csv(file) %>% 
    select(-any_of(c('Name', 'Habitat', 'Networked_bacteria'))) %>% 
    mutate(Habitat = str_to_title(x)) %>% 
    distinct()
})

micro_data_df <- do.call(rbind, micro_data) %>% 
  mutate(type = 'Bacterial abundance') %>% 
  select(cluster,
         var = Microbe,
         rho,
         pval_adj = p_adj,
         type,
         Habitat)

genome_belong <- micro_data_df %>% 
  filter(cluster == 'Cluster1',
         Habitat == 'Bog') %>% 
  select(var, Habitat) %>% 
  distinct() %>% 
  mutate(Habitat = factor(Habitat, levels = c('Palsa', 'Bog', 'Fen')))
```

## MetaT annotation - Expression at pathway level

```{r}
metaT_exp <- read_csv('../input/Emerge_metaTs_processed_v7/metaT_pathway_exp_filter_2012.csv') %>% 
  filter(str_detect(SampleID__, '2012[0-9]+_S_'))

```

# Changing pathway name format
```{r}
metaT_exp2 <- metaT_exp %>% 
  mutate(subpathway = str_remove_all(subpathway, '-all'),
         subpathway = str_replace_all(subpathway, '_', ' '),
         pathway = str_replace_all(pathway, '_', ' '))

```


## MetaT annotation - per gene

```{r}
meta_path <- read_csv('../input/Emerge_metaTs_processed_v7/metaT_pathway_filter_2012.csv') %>% 
  filter(str_detect(SampleID__, '2012[0-9]+_S_')) %>% 
  select(!c(contig, start, end, strand, length, gene, product, temporal_sample_id))

# Cazy

metaT_path_cazy <- meta_path %>% 
  filter(pathway == "CAZy")


```



## OTU table

```{r}
# Get list of annotated genomes
genomes <- genome_belong %>% 
  pull(var) %>% 
  unique()

otu_table <- read_csv("../Assembly_feature_level_microbiome/input/otu_normalized.csv") %>% 
  filter(genome %in% genomes) %>% 
  separate_wider_delim(GTDB_v2_r207, delim = ";", names = c('Domain', 'Phylum',
                                                            'Class', 'Order', 'Family',
                                                            'Genus','Species')) %>% 
  select(genome, all_of(metadata$names_bnti), Phylum) %>% 
  pivot_longer(!c(genome, Phylum), names_to = 'sampleid', values_to = 'norm_abundance') %>% 
  left_join(metadata, by = c('sampleid' = 'names_bnti')) %>% 
  group_by(genome, Habitat, Phylum) %>% 
  summarise(summed_norm_abundance = sum(norm_abundance)) %>% 
  inner_join(genome_belong, by = c('genome' = 'var', 'Habitat')) %>% 
  mutate(Phylum = factor(Phylum)) %>% 
  ungroup() %>% 
  arrange(Phylum) %>% 
  mutate(order = n():1)
```

# Supplementary figure 2

## Pathway expression version ##

Expression of genomes that correlated with cluster 1 of the bog

```{r}

metaT_annot_ready <- metaT_exp2 %>% 
  filter(genome %in% genome_belong$var) %>% 
  inner_join(genome_belong, by = c('genome' = 'var')) %>% 
  inner_join(otu_table, by = c('genome', 'Habitat'))


supp2 <- metaT_annot_ready %>% 
  group_by(SampleID__, pathway, subpathway, Phylum) %>% 
  summarise(tpm = sum(tpm)) %>% 
  mutate(Phylum = str_remove(Phylum, 'p__')) %>% 
  ggplot() +
  geom_tile(aes(x = SampleID__,
                y = subpathway,
                fill = tpm),
            color = 'white') +
  theme_bw() +
  facet_grid(rows = vars(pathway),
             cols = vars(Phylum),
             scales = 'free',
             space = 'free') +
  scale_x_discrete(expand = c(0, 0)) +
  scale_y_discrete(expand = c(0, 0)) +
  #scale_fill_distiller(palette = 'YlOrRd', direction = 1) +
  scale_fill_viridis_c(option = 'A', direction = -1)+
  labs(fill = "TPM")+
  theme(axis.title = element_blank(),
        axis.text.x = element_blank(),
        axis.text.y = element_text(size = 6),
        strip.text.y = element_text(angle = 0, size = 6),
        strip.text.x = element_text(angle = 90, size = 6),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        legend.position = "bottom",
        legend.text = element_text(size = 6))  

supp2

ggsave('output_updated/Supplementary_fig_2.png', supp2, dpi = 300, width = 180, units = "mm")

ggsave('output_updated/Supplementary_fig_2.eps', supp2, 
       device = cairo_ps, fallback_resolution = 300,
       width = 180, height = 110, units = 'mm')



```


# Supplementary figure 3

Expression of genomes that correlated with greenhouse gases of the bog

```{r}
selected_genomes <- c('20120700_S1D_25', '20110800_S2M_3',
                      '20120800_E1D_14', '20140700_E14_11',
                      '20120600_S1S_4')

metaT_filtered <- metaT_exp2 %>% 
  filter(genome %in% selected_genomes)

label_names <- set_names(c('Acidobacteriota<br>*Holophaga*<br>(20120700_S1D_25)',
                           'Acidobacteriota<br>*Terracidiphilus*<br>(20110800_S2M_3)',
                           'Acidobacteriota<br>*Terracidiphilus*<br>(20120800_E1D_14)',
                           'Actinobacteriota<br>*Fen-455*<br>(20140700_E14_11)',
                           'Actinobacteriota<br>*RAAP-2*<br>(20120600_S1S_4)'),
                         nm = selected_genomes)

supp3 <- metaT_filtered %>% 
  mutate(genome = factor(genome, levels = selected_genomes)) %>% 
  ggplot() +
  geom_tile(aes(x = SampleID__,
                y = subpathway,
                fill = tpm),
            color = 'white') +
  theme_bw() +
  facet_grid(
             cols = vars(genome),
             scales = 'free',
             space = 'free',
             labeller = labeller(genome = label_names)) +
  labs(fill = "TPM")+
  scale_x_discrete(expand = c(0, 0)) +
  scale_y_discrete(expand = c(0, 0)) +
  scale_fill_viridis_c(option = 'A', direction = -1)+
  theme(axis.title = element_blank(),
        axis.text.x = element_blank(),
        strip.text.x = element_markdown(size = 6, angle = 90),
        axis.text.y = element_text(size = 6),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        legend.position = "bottom",
        legend.title = element_text(size = 6),
        legend.text = element_text(size = 6))  

supp3

```

## CAZY
```{r}

metaT_annot_cazy <- metaT_path_cazy %>% 
  filter(genome %in% selected_genomes) %>% 
  drop_na(cazy_id) %>% 
  mutate(cazy_fam = case_when(
    str_detect(cazy_id, 'GH') ~ 'Glycoside\nHydrolases',
    str_detect(cazy_id, 'GT') ~ 'Glycosyl\nTransferases',
    str_detect(cazy_id, 'PL') ~ 'Polysaccharide\nLyases',
    str_detect(cazy_id, 'CE') ~ 'Carbohydrate\nEsterases',
    str_detect(cazy_id, 'AA') ~ 'Auxiliary Activities',
    str_detect(cazy_id, 'CBM') ~ 'Carbohydrate\nBinding Module'))

supp3.1 <- metaT_annot_cazy %>% 
  group_by(SampleID__, pathway, label_id, genome, cazy_fam) %>% 
  summarise(tpm = sum(tpm)) %>% 
  ggplot() +
  geom_tile(aes(x = SampleID__,
                y = label_id,
                fill = tpm),
            color = 'white') +
  theme_bw() +
  facet_grid(rows = vars(cazy_fam),
             cols = vars(genome),
             scales = 'free',
             space = 'free',
             labeller = labeller(genome = label_names)) +
   labs(fill = "TPM")+
  scale_x_discrete(expand = c(0, 0)) +
  scale_y_discrete(expand = c(0, 0)) +
  scale_fill_viridis_c(option = 'A', direction = -1)+
  theme(axis.title = element_blank(),
        axis.text.x = element_blank(),
        axis.text.y = element_text(size = 6),
        strip.text.y = element_text(angle = 0, size = 6),
        strip.text.x = element_markdown(size = 6, angle = 90),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        legend.position = "bottom",
        legend.title = element_text(size = 6),
        legend.text = element_text(size = 6)) 

supp3.1

## Joining figures


final <- ggarrange(supp3, supp3.1, labels = 'AUTO')


final

ggsave('output_updated/Supplementary_fig_3.png', final, dpi = 300, width = 180, height = 150, units = "mm")



ggsave('output_updated/Supplementary_fig_3.eps',final, 
       device = cairo_ps, fallback_resolution = 300,
       width = 180, height = 150, units = 'mm')


```









